### PR TITLE
Windpower Module Example

### PR DESCRIPTION
**Pull request by cpaulgilman**
Python code and supporting files for a Windpower module example:

* Use a wind resource file from WIND Toolkit API
* Sample JSON file for wind turbine parameters
* Wind farm layout with single turbine or multiple turbines
* Readme with explanation of how to work with turbine coordinates and get wind farm layout from SAM UI to PySAM

Addresses questions on PySAM forum: https://sam.nrel.gov/forum/forum-sdk/4218

### CODE REVIEW

File: stubs/stubs/AdjustmentFactors.pyi
Comment (dguittet) [inline stubs/stubs/AdjustmentFactors.pyi:1]: Did you mean to delete this?

File: files/sscd.lib
Comment (dguittet) [inline files/sscd.lib:1]: Did you mean to add this?

File: Examples/Windpower Example/windpower-example.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
+import json
+import PySAM.Windpower as wp
+
+# create a new windpower model
+wind = wp.new()
+
+# set model inputs from JSON generated by SAM
+with open('windpower-inputs.json', 'r') as file:
```
Comment (dguittet) [inline Examples/Windpower Example/windpower-example.py:8]: You should use the path library to make sure the input json can be found regardless of the working directory

```
from pathlib import Path

Path(__file__).parent / 'windpower-inputs.json`
```

Comment (dguittet) [inline Examples/Windpower Example/windpower-example.py:8]: You should use the path library to make sure the input json can be found regardless of the working directory

```
from pathlib import Path

Path(__file__).parent / 'windpower-inputs.json`
```

File: Examples/Windpower Example/readme.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# PySAM Windpower Module Example
+
+This example illustrates how to work with PySAM's Windpower module for wind turbines and wind farms.
+
```
Comment (dguittet) [inline Examples/Windpower Example/readme.md:4]: Can you add what version of PySAM and SAM this was created with, and what it was last tested with?

File: Examples/Windpower Example/windpower-example.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
+import json
```
Comment (dguittet) [inline Examples/Windpower Example/windpower-example.py:1]: Can you add what version of PySAM and SAM this was created with, and what it was last tested with?

File: Examples/Windpower Example/windpower-example.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+## replace some inputs as needed for this analysis ##
+
+# load a wind resource file
+wind.value(	'wind_resource_filename', 'wind-resource-file.csv')
+
+# wind shear is location-specific to adjust wind speed data to turbine hub height
+wind.value('wind_resource_shear', 0.14)
+
+# load wind turbine parameters from JSON
+with open('wind-turbines.json', 'r') as file:
```
Comment (dguittet) [inline Examples/Windpower Example/windpower-example.py:24]: same with this file path

File: Examples/Windpower Example/windpower-example.py [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    sam_data = json.load(file)
+    # loop through each key-value pair
+    for k, v in sam_data.items():
+        if k != 'number_inputs':
+            wind.value(k, v)
+
+## replace some inputs as needed for this analysis ##
+
+# load a wind resource file
+wind.value(	'wind_resource_filename', 'wind-resource-file.csv')
```
Comment (dguittet) [inline Examples/Windpower Example/windpower-example.py:18]: `wind.value( 'wind_resource_filename', str(Path(__file__).parent / 'wind-resource-file.csv'))`

**Review by dguittet — CHANGES_REQUESTED** [review]

File: Examples/Windpower Example/windpower-example.py (hunk 4 above)
Comment (cpaulgilman) [reply Examples/Windpower Example/windpower-example.py]: Fixed with 62f40619128218a81b30029a22a925f2dfb43a5d

File: Examples/Windpower Example/windpower-example.py (hunk 5 above)
Comment (cpaulgilman) [reply Examples/Windpower Example/windpower-example.py]: Fixed with 62f40619128218a81b30029a22a925f2dfb43a5d

File: files/sscd.lib
Comment (cpaulgilman) [reply files/sscd.lib:1]: Fixed in bf2b33e51bfa03826128b879d93585ad3612a795

File: stubs/stubs/AdjustmentFactors.pyi
Comment (cpaulgilman) [reply stubs/stubs/AdjustmentFactors.pyi:1]: Fixed in d61743a460064d8ba3a907e7f71e9fcd3bc8ca8a

File: Examples/Windpower Example/readme.md (hunk 2 above)
Comment (cpaulgilman) [reply Examples/Windpower Example/readme.md:4]: Fixed in f4149970cc3a4f3134c0a4f51c542baf1af9cce6

**Review by dguittet — APPROVED** [review]
LGTM Thanks for the changes!

**Comment by cpaulgilman** [comment]
@dguittet I did this work in the pysam/dev branch. Should I merge it into pysam/main now?

**Comment by dguittet** [comment]
Yes please

**Pull request merged**